### ISSUE TITLE
Add Group policy for playlist

### ISSUE THREAD

**Issue by kdenhartog** [brave/brave-browser#41428]
### Platforms

Linux, macOS, Windows

### Description

In https://github.com/brave/brave-core/pull/25710 we started to migrate towards group policies that match how Chromium does it. We currently have a few group policies for Brave specific features and this issue is to add a new policy that would disable playlist as well.

**Comment by bsclifton** [comment]
Folks who were interested in this - you can check out the steps outlined here:
https://github.com/brave/brave-core/tree/master/components/policy/resources/templates/policy_definitions

Those go over the steps needed to create a new policy. You can also search our pull requests for `group policy` to find some of the prior implementations folks have done.

**Comment by deepthiteki** [comment]
Please assign this to me

**Comment by brokoli777** [comment]
Hi! If this feature hasn't been completed, can I work on it? I don't need to be assigned to it.

**Comment by kdenhartog** [comment]
Certainly, I've not seen a PR for it yet so this is still free for anyone to work on.

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.84.114 Chromium: 141.0.7390.55 (Official Build) beta (64-bit)
-- | --
Revision | 599996719ea803b88c14619421f85a7575cc2e8d
OS | Windows 11 Version 24H2 (Build 26100.6584)
```

Verified test plan from https://github.com/brave/brave-core/pull/30351#issue-3275336222

<details>
<summary>Case 0: Add Windows Group policy templates via policy_templates.zip - PASSED </summary>

1. Installed `1.82.144`
2. launched Brave
3. opened https://github.com/brave/brave-browser/releases/download/v1.82.144/policy_templates.zip in a new tab
5. downloaded and extracted `policy_templates.zip file
6. closed Brave
7. opened Local Group Policy editor (Windows>>Run>>enter gpedit.msc>>OK)
8. right clicked Administrative Templates under Computer Configuration
9. selected `Add` in the  Add/Remove templates
10. navigated to Downloads> policy_templates > windows > adm > en-US > brave.adm > open
11. closed Add/Remove Templates dialog
12. navigated to Local Group Policy Editor -->Administrative Templates
13. expanded Classic Administrative Templates (ADM) -->Brave Software -->Brave -->Brave software settings
14. verified `Enable Brave Playlist` entry shows `Not configured`
 
step 12| step 13
------ | -------
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/2381baef-f68a-4362-b428-f7f2f40f0959" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/73e99e23-1f2b-4cc4-82cb-3e64ce328d96" />

</details>

</details>


<details>
<summary>Case 1: Disable Brave Playlist via group policy - PASSED</summary>
Pre-requisite: Enable `#playlist` via `brave://flags`

1. continued from case 1 (step 14)
2. selected `Disabled` ---> Apply --> Ok in the Brave software settings for `Enable Brave Playlist`
3. opened Brave
4. opened youtube.com and started playing a video
5. confirmed `playlist` button not shown in the sidebar and URL bar
6. navigated to `brave://policy`
7. confirmed that `BravePlaylistEnabled` is listed as "false"
8. searched for `playlist`
9. added a shortcut `Alt + P` 
10. returned to youtube.com tab
11. pressed `Alt + P` 
12. confirmed `playlist` bubble is not shown
13. opened `brave://settings/braveContent`
14. confirmed `Playlist` toggle is shown disabled and read only.
15. confirmed `browser is managed by your organiztion` icon is shown
16. closed Brave

 step 2 | step 4 | step 5 | step 8 | step 14
------- | ------- | ------ | --------- | ------ 
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/8c58aa1b-c679-4afa-b6e9-2aa7cc17595d" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/f5ed5604-475a-4e7a-a81b-76d19b87539c" />|<img width="1921" height="492" alt="Image" src="https://github.com/user-attachments/assets/c06ec9bf-5ad5-448d-b8f7-ff6b5e6247c2" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/3f7d3123-588a-4339-b7fe-22b543cc315e" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/fcac0eaa-8142-4a94-8af0-e86c7ccb070e" />

</details>

<details>
<summary>Case 2:  Enable Brave Playlist via group policy - PASSED</summary>

1. Continued from Case 1 
2. opened `Local Group Policy Editor`
3. selected `Enabled` --> Apply --> Ok in the Brave software settings for `Enable Brave Playlist`
4. relaunched Brave
5. returned to youtube.com tab
6. confirmed `playlist` button shown in the sidebar and URL bar
6. navigated to `brave://policy` 
7. confirmed that `BravePlaylistEnabled` is listed as "true"
8. opened `brave://settings/system/shortcuts`
9. search for `playlist`
10. returned to youtube.com tab
11. pressed `Alt + P` 
12. confirmed `playlist` bubble is shown
13. opened `brave://settings/braveContent`
14. confirmed `Playlist` toggle is shown enabled but read only.
15. confirmed `browser is managed by your organiztion` icon is shown
16. confirmed `Automatically add files for `offine` viewing setting is shown
17.  closed Brave

step 3 | step 4 | step 6 | step 7 | step 8 | step 12 | step 16
------ | ------- | --------- | -------- | -------- | -------|------- 
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/26afb67a-2549-4342-a967-fa8257eb3e63" />|<img width="1914" height="398" alt="Image" src="https://github.com/user-attachments/assets/57114666-6a27-4e06-931a-8bc3d401a66e" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/4b0e0563-4ce6-483c-8fe5-1a29691e40a8" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/89e78015-3675-4871-8ef0-bd56ec248316" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/cc796178-683b-4816-9a9b-e81a135c7e73" />|<img width="1914" height="485" alt="Image" src="https://github.com/user-attachments/assets/6cf99f5b-82c6-4d52-853f-34300f8cf51b" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/97649fe3-8d3d-4519-81c6-1733d422963d" />

</details>

<details>
<summary>Case 3:  Set `Brave Playlist` to `Not configured` via group policy - PASSED</summary>

1. Continued from Case 2
2. opened `Local Group Policy Editor`
3. selected `Not configured` --> `Apply` --> `Ok` in the Brave software settings for `Enable Brave Playlist`
4. relaunched Brave
5. navigated to `brave://policy` 
6. confirmed that `Enable Brave Playlist` is no longer listed
7. confirmed Playlist button is still shown in the URL bar and sidebar as expected
8. confirmed Playlist is enabled and `browser is managed by your organization` icon is not shown
9. confirmed `Automatically add files for `offine` viewing setting is shown
10. closed Brave

step 3 | step 5 | step 7 | step 8-9
------ | ------- | ------- | ------
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/c46e715d-3dbf-4fda-88e7-41849f167be7" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/8f33cf29-adb4-4614-ac72-7954f0116626" />|<img width="1918" height="474" alt="Image" src="https://github.com/user-attachments/assets/dbb5d9da-5238-468c-ad8f-99bea6b5b2ce" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/d26e9334-72bc-4252-aacb-4c73773bc91d" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/56a15da7-4cba-49a5-ab61-a9a5def48819" />

</details>
</details>
